client/packages/invoices/src/StockOut/utils.ts

```diff
@@ -169,15 +169,23 @@ export const canAllocate = (line: DraftStockOutLineFragment): boolean =>
 
 export const canAutoAllocate = (
   line: DraftStockOutLineFragment,
+  expiryThresholdDays: number | undefined,
   requiredPackSize?: number
-) =>
-  canAllocate(line) &&
-  // shouldn't auto-allocate expired lines
-  !(!!line.expiryDate && DateUtils.isExpired(new Date(line.expiryDate))) &&
-  // should not be able to auto-allocate lines with unusable VVM status
-  !line.vvmStatus?.unusable &&
-  // if pack size is specified, should match the line's pack size
-  (!requiredPackSize || line.packSize === requiredPackSize);
+) => {
+  const lastAllowableDate = line.expiryDate
+    ? DateUtils.addDays(new Date(line.expiryDate), -(expiryThresholdDays ?? 0))
+    : undefined;
+
+  return (
+    canAllocate(line) &&
+    // shouldn't auto-allocate expired lines
+    !(!!lastAllowableDate && DateUtils.isExpired(lastAllowableDate)) &&
+    // should not be able to auto-allocate lines with unusable VVM status
+    !line.vvmStatus?.unusable &&
+    // if pack size is specified, should match the line's pack size
+    (!requiredPackSize || line.packSize === requiredPackSize)
+  );
+};
```

Comment:
Since we have this calculation here, I was going to put it in its own function and re-use it in the components for row disabling. However, they're slightly different -- this one always should get `expiryThresholdDays = 0` if the pref is off, because it still needs to avoid auto-allocation when expired regardless of the pref (that's its current behaviour), whereas the line-disabling components should only look at the threshold number if the preference is actually enabled, so it doesn't worry about expiry at all if not.

Hope that makes sense 🤔